### ISSUE TITLE
Training data for 16x

### ISSUE THREAD

**Issue by juhha1** [EasyRy/RepKPU#2]
Thanks for the interesting project!
I am trying to replicate the work for 16x upsampling experiment, but I cannot find the data or dataset generation code.

How does training set generated?

Thanks!

**Comment by EasyRy** [comment]
Thanks for your interest for our work! 
The 16x training set is the same as 4x, that is, the model only has weights for 4x upsamling. However, in the testing phase, 16x upsampling can be achieved by conducting 4x upsampling twice.

**Comment by juhha1** [comment]
Got it. Thanks for your quick response!

I have two more questions.
1. I want to clarify my understanding for this. For 16x, we can run 4x twice. For other arbitrary scaling factors such as 5x and 19x, F2A is applied on top of the upsampled output from the mode. Am I understanding correctly?
2. Is there any code making a dataset? Like for PU-GAN, there is a training samples in .h5. When I want to run this model on other point-cloud dataset such as KITTI, or any other, is there a way to generate the training samples?

Agian, thanks x100!

**Comment by EasyRy** [comment]
You are welcome!
1. F2A is an interpolation-then-upsampling method. We first interpolate the origin point cloud, and then feed the interpolated result into model for upsampling, as I think the interpolation result is noisy and needs a network to refine it.
2. It's an interesting idea to build a new dataset. Actually, I think PU-GAN and PU1K are both too small to reflect the real model performances. The PU-GAN and PU1K both consist of patches instead of complete shapes, if you want to generate a patch-based dataset, you can refer to the corresponding papers or generate by yourself, e.g., select several 'seed' points and extract the neighbor points around them to form patches. If you want to generate complete shapes, you can refer to [here](https://github.com/wangzy22/TAP) to extract point clouds from mesh files.

**Comment by juhha1** [comment]
Thank you very much for your answer!
